src/controllers/account/account.ts

```diff
@@ -17,35 +22,75 @@ export const generateResetToken = async (): Promise<ResetToken> => {
   return { selector, verifier }
 }
 
-export const getEmailResetToken = async (input: ResetToken): Promise<string> => {
+export const getEmailAuthToken = async (input: AuthToken): Promise<string> => {
   return Buffer.concat([input.selector, input.verifier]).toString('base64')
 }
 
-export const getVerifierHash = async (input: ResetToken): Promise<Buffer> => {
+export const getVerifierHash = async (input: AuthToken): Promise<Buffer> => {
   const verifierHash = Buffer.alloc(sodium.crypto_generichash_BYTES)
   sodium.crypto_generichash(verifierHash, input.verifier)
   return verifierHash
 }
 
-// public static async storeResetToken (selector: Buffer, verifierHash: Buffer) {
-//   // TODO store selector in DB along with hash(verifier)
-// }
+export const storeAuthToken = async (email: string, role: string, selector: Buffer, verifierHash: Buffer) => {
+  // Store email and selector in dynamo DB instance along with hash(verifier)
+  var params = {
+    Item: {
+      email, 
+      role,
+      selector,
+      verifierHash
+    }, 
+    TableName: TABLE_NAME
+  }
+  dynamo.putItem(params, (err: Error, data: {}) => {
+    if (err) console.log(err, err.stack)
+    else {
+      console.log(data)
+    }    
+  })
+}
 
-export const splitSelectorVerifier = async (token: string): Promise<ResetToken> => {
+export const splitSelectorVerifier = async (token: string): Promise<AuthToken> => {
   const tokenBuf = Buffer.from(token, 'base64')
   const selector = tokenBuf.slice(0, 16)
   const verifier = tokenBuf.slice(16)
   return { selector, verifier }
 }
 
-// public static async deleteSelector (resetToken: ResetToken) {
-//   // TODO delete resetToken.selector from DB
-// }
+export const deleteSelector = async (authToken: AuthToken) => {
+  // Delete authToken.selector from dynamo db instance
+  const params = {
+    Key: {
+     selector: authToken.selector
+    }, 
+    TableName: TABLE_NAME
+  }
+  dynamo.deleteItem(params, (err: Error, _: any) => {
+    if (err) console.log(err, err.stack)
+    else {
+      console.log('deleted', authToken)
+    }
+  })
+}
 
-// public static async getStoredVerifierHash (resetToken: ResetToken): Promise<Buffer> {
-//   // TODO lookup resetToken.selector in DB and return stored verifier hash
-//   return Buffer.alloc(0)
-// }
+export const getStoredVerifierHash = async (authToken: AuthToken): Promise<Buffer> => {
+  // Lookup resetToken.selector in DB and return stored verifier hash
+  const params = {
+    Key: {
+     selector: authToken.selector
```

Comment:
have these Dynamo queries been tested? usually you either need to use the [DocumentClient](https://docs.aws.amazon.com/AWSJavaScriptSDK/latest/AWS/DynamoDB/DocumentClient.html) to make JS types Dynamo-friendly, or else use the [Converter](https://docs.aws.amazon.com/AWSJavaScriptSDK/latest/AWS/DynamoDB/Converter.html) to do the same thing

Reply:
awesome good call